diff --git a/Cargo.lock b/Cargo.lock
index 38f347286..531d3d042 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -4288,6 +4288,7 @@ dependencies = [
  "wasm-bindgen-futures",
  "web-sys",
  "winreg 0.10.1",
+ "zstd",
 ]
 
 [[package]]
@@ -6058,4 +6059,5 @@ checksum = "9fd07cbbc53846d9145dbffdf6dd09a7a0aa52be46741825f5c97bdd4f73f12b"
 dependencies = [
  "cc",
  "libc",
+ "pkg-config",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 799dfb6bb..f9861e85f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,6 +68,8 @@ is_executable = "1.0.1"
 simplelog = "0.12.0"
 time = "0.3.12"
 
+zstd = { version = "*", features = [ "pkg-config" ] }
+
 [target.'cfg(not(target_os = "windows"))'.dependencies]
 # Our dependencies don't use OpenSSL on Windows
 openssl = { version = "0.10.38", features = ["vendored"], optional = true }
